### ISSUE TITLE
Unexpected Interference result for dict literal containing lists

### ISSUE THREAD

**Issue by moosbruggerj** [microsoft/pyright#10675]
I am not sure if this is actually a bug or intended behavior, since pyright is technically correct. However, the result is unexpected to me.

Pyright is able to interfere the type of a dict, if all keys/values can be inferred and they match, which is great. If the types mismatch, it falls back to unknown, which is also expected. However, since an empty list has type `list[Unknown]`, having empty and non empty lists in a dictionary leads to an unexpectedly underspecified type.

**Code or Screenshots**

```python
a = {"a": ["a"], "b": ["b"]}
reveal_type(a)  # Type of "a" is "dict[str, list[str]]"
b = {"a": ["a"], "b": []}
reveal_type(b)  # Type of "b" is "dict[str, Unknown]"
```

I would expect `b` to be of type `dict[str, list[str]]` as well.
Adding explicit type hints for the variables works, but it adds noise to the code IMHO.

Not sure, if this is relevant, but mypy (1.16.1) handles the special case of empty lists like I would expect it.
```
test.py:2: note: Revealed type is "builtins.dict[builtins.str, builtins.list[builtins.str]]"
test.py:4: note: Revealed type is "builtins.dict[builtins.str, builtins.list[builtins.str]]"
```
For distinct types mypy has the same behavior as pyright and falls back to object.

**VS Code extension or command-line**
pyright lsp: 1.1.402 (via nvim-lsp)

**Comment by erictraut** [comment]
Pyright is working as intended here.

For information on how pyright infers dictionary and list literals, refer to [this documentation](https://microsoft.github.io/pyright/#/type-inference?id=dictionary-expressions).

Regardless of the inference rules that a type checker employs, there will always be cases where the results differ from the intent of the programmer. In those cases, you can use an explicit type annotation for the target variable, and pyright will use [bidirectional type inference](https://microsoft.github.io/pyright/#/type-inference?id=bidirectional-type-inference-expected-types) to honor that type.
